api/v1/routes.go

```diff
@@ -22,27 +22,80 @@ package v1
 
 import (
 	"net/http"
-
-	"github.com/julienschmidt/httprouter"
 )
 
 func NewHandler() http.Handler {
-	router := httprouter.New()
+	mux := http.NewServeMux()
+
+	mux.HandleFunc("/v1/status", func(rw http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodGet:
+			handleGetStatus(rw, r)
+		case http.MethodPatch:
+			handlePatchStatus(rw, r)
+		default:
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+		}
+	})
+
+	mux.HandleFunc("/v1/metrics", func(rw http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		handleGetMetrics(rw, r)
+	})
+
+	mux.HandleFunc("/v1/metrics/", func(rw http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		id := r.URL.Path[len("/v1/metrics/"):]
+		handleGetMetric(rw, r, id)
+	})
+
+	mux.HandleFunc("/v1/groups", func(rw http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+
+		handleGetGroups(rw, r)
+	})
 
-	router.GET("/v1/status", HandleGetStatus)
-	router.PATCH("/v1/status", HandlePatchStatus)
+	mux.HandleFunc("/v1/groups/", func(rw http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
 
-	router.GET("/v1/metrics", HandleGetMetrics)
-	router.GET("/v1/metrics/:id", HandleGetMetric)
+		id := r.URL.Path[len("/v1/groups/"):]
+		handleGetGroup(rw, r, id)
+	})
 
-	router.GET("/v1/groups", HandleGetGroups)
-	router.GET("/v1/groups/:id", HandleGetGroup)
+	mux.HandleFunc("/v1/setup", func(rw http.ResponseWriter, r *http.Request) {
+		switch r.Method {
+		case http.MethodPost:
+			handleRunSetup(rw, r)
+		case http.MethodPut:
+			handleSetSetupData(rw, r)
+		case http.MethodGet:
+			handleGetSetupData(rw, r)
+		default:
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+		}
+	})
 
-	router.POST("/v1/setup", HandleRunSetup)
-	router.PUT("/v1/setup", HandleSetSetupData)
-	router.GET("/v1/setup", HandleGetSetupData)
+	mux.HandleFunc("/v1/teardown", func(rw http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodPost {
+			rw.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
 
-	router.POST("/v1/teardown", HandleRunTeardown)
+		handleRunTeardown(rw, r)
+	})
 
-	return router
+	return mux
```

Comment:
I really do wonder if I shouldn't just make one big switch with `string.HasPrefix` and friends?

Reply:
:+1: it might be a bit better, but even like this it's a great improvement compared to before, so you decide :man_shrugging:

Reply:
I started doing it and looks worse IMO so I am going to just leave it